pkg/nvml/gpm.go

```diff
@@ -57,14 +57,23 @@ type GpmMetricsGetVType struct {
 func (l *library) GpmMetricsGetV(metricsGet *GpmMetricsGetType) GpmMetricsGetVType {
 	return GpmMetricsGetVType{metricsGet.convert()}
 }
+
+// nvmlGpmMetricsGetStub is a stub function that can be overridden for testing.
+var nvmlGpmMetricsGetStub = nvmlGpmMetricsGet
+
 func (metricsGetV GpmMetricsGetVType) V1() Return {
 	metricsGetV.metricsGet.Version = 1
-	return nvmlGpmMetricsGet(metricsGetV.metricsGet)
+	return nvmlGpmMetricsGetStub(metricsGetV.metricsGet)
 }
 
 func (l *library) GpmMetricsGet(metricsGet *GpmMetricsGetType) Return {
 	metricsGet.Version = GPM_METRICS_GET_VERSION
-	return nvmlGpmMetricsGet(metricsGet.convert())
+	nvmlMetricsGet := metricsGet.convert()
+	defer func() {
+		metricsGet.Metrics = nvmlMetricsGet.Metrics
+	}()
+
+	return nvmlGpmMetricsGetStub(nvmlMetricsGet)
```

Comment:
I don't like the idea of adding a one-off stub like this. If we are going to do this, we should do it for everything.

Reply:
To be clear, I added one like this for the topology functions that I also added tests for. I didn't see the value in adding all of them if we weren't going to use them immediately.

Would you rather I remove the tests and the stubs?

Reply:
I guess it's fine then. I think we can add a level of indirection pretty easily though. We just instruct `nvidia.yml` to rewrite all generated functions as `nvml*Internal` and then do something similar to how I generate the various `interfaces` to generate variables for all `nvml* = nvml*Internal` by default. Even better would be to make them all function pointers hanging off of the `library` type (so we don't have to do the weird update/reset in the tests), but that might be more trouble than it's worth.

Reply:
Sure. Let's have a look at this at some point in the (near) future.